exonum/src/blockchain/tests.rs

```diff
@@ -362,7 +390,117 @@ fn create_blockchain(
+fn create_blockchain(
+    services: Vec<Box<dyn ServiceFactory>>,
+    instances: Vec<impl Into<InstanceInitParams>>,
+) -> BlockchainMut {
+    maybe_create_blockchain(services, instances).unwrap()
+}
+
+/// Checks that `Schema::height` and `Schema::next_height` work as expected.
+#[test]
+fn blockchain_height() {
+    let mut blockchain = create_blockchain(
+        Vec::<Box<dyn ServiceFactory>>::new(),
+        Vec::<InstanceInitParams>::new(),
+    );
+
+    // Check that height is 0 after genesis creation.
+    let snapshot = blockchain.snapshot();
+    let schema = Schema::new(&snapshot);
+    assert_eq!(schema.height(), Height(0));
+    assert_eq!(schema.next_height(), Height(1));
+
+    // Create one block.
+    let (_, patch) = blockchain.create_patch(
+        ValidatorId::zero(),
+        Height::zero(),
+        &[],
+        &mut BTreeMap::new(),
+    );
+    blockchain.merge(patch).unwrap();
+
+    // Check that height is 1.
+    let snapshot = blockchain.snapshot();
+    let schema = Schema::new(&snapshot);
+    assert_eq!(schema.height(), Height(1));
+    assert_eq!(schema.next_height(), Height(2));
+}
+
+/// Checks that before genesis creation `Schema::height` panics.
+#[test]
+#[should_panic(
+    expected = "An attempt to get the actual `height` during creating the genesis block"
+)]
+fn blockchain_height_panics_before_genesis() {
+    let service_keypair = crypto::gen_keypair();
+
+    // Create a blockchain *without* creating a genesis block.
+    let blockchain = Blockchain::new(TemporaryDB::new(), service_keypair, ApiSender::closed());
+
+    let snapshot = blockchain.snapshot();
+    let schema = Schema::new(&snapshot);
+    let _height = schema.height();
+}
+
+/// Checks that before genesis creation `Schema::next_height` doesn't panic.
+#[test]
+fn blockchain_next_height_does_not_panic_before_genesis() {
+    let service_keypair = crypto::gen_keypair();
+
+    // Create a blockchain *without* creating a genesis block.
+    let blockchain = Blockchain::new(TemporaryDB::new(), service_keypair, ApiSender::closed());
+
+    let snapshot = blockchain.snapshot();
+    let schema = Schema::new(&snapshot);
+    let height = schema.next_height();
+    assert_eq!(height, Height(0))
+}
+
+/// Checks that `after_transactions` is invoked for services added
+/// within genesis block.
+#[test]
+fn after_transactions_invoked_on_genesis() {
+    // `ServiceGoodImpl` sets the value in schema within `after_transactions`.
+    let blockchain = create_blockchain(
+        vec![ServiceGoodImpl.into()],
+        vec![ServiceGoodImpl
+            .artifact_id()
+            .into_default_instance(3, "service_good")],
+    );
+
+    // After creation of the genesis block, check that value was set.
+    let snapshot = blockchain.snapshot();
+    let index = snapshot.get_list("service_good.val");
+    assert_eq!(index.len(), 1);
+    assert_eq!(index.get(0), Some(1));
+}
+
+/// Checks that if `after_transactions` fails on the genesis level,
```

Comment:
Nit: I'd reword "on the genesis level" to "on the genesis block" here and elsewhere in order to unify the used terms.